frontend/ui/widget/statuswidget.lua

```diff
@@ -0,0 +1,653 @@
+        face = self.small_font_face,
+    }
+
+    local progress_bar_text_container = CenterContainer:new {
+        dimen = Geom:new { w = width, h = text_complete:getSize().h },
+        text_complete
+    }
+
+    table.insert(title_author_progressbar_group, progress_bar_text_container)
+    table.insert(title_author_container, title_author_progressbar_group)
+    return title_author_container
+end
+
+
+function StatusWidget:onAnyKeyPressed()
+    return self:onClose()
+end
+
+function StatusWidget:onClose()
+    self:saveSummary()
+    UIManager:setDirty("all")
+    UIManager:close(self)
+    return true
+end
+
+
+--TODO: MOVE TO UTILS AND CHANGE ALSO TEXTBOXWIDGET
+function StatusWidget:_wrapGreedyAlg(h_list, width)
+    local cur_line_width = 0
+    local cur_line = {}
+    local v_list = {}
+
+    for k, w in ipairs(h_list) do
+        w.box = {
+            x = cur_line_width,
+            w = w.width,
+        }
+        cur_line_width = cur_line_width + w.width
+        if w.word == "\n" then
+            if cur_line_width > 0 then
+                -- hard line break
+                table.insert(v_list, cur_line)
+                cur_line = {}
+                cur_line_width = 0
+            end
+        elseif cur_line_width > width then
+            -- wrap to next line
+            table.insert(v_list, cur_line)
+            cur_line = {}
+            cur_line_width = w.width
+            table.insert(cur_line, w)
+        else
+            table.insert(cur_line, w)
+        end
+    end
+    -- handle last line
+    table.insert(v_list, cur_line)
+
+    return v_list
+end
+
+--TODO: MOVE TO UTILS AND CHANGE ALSO TEXTBOXWIDGET
+function StatusWidget:_getVerticalList(text, width, face, bold)
+    -- build horizontal list
+    local h_list = {}
+    for line in util.gsplit(text, "\n", true) do
+        for words in line:gmatch("[\32-\127\192-\255]+[\128-\191]*") do
+            for word in util.gsplit(words, "%s+", true) do
+                for w in util.gsplit(word, "%p+", true) do
+                    local word_box = {}
+                    word_box.word = w
+                    word_box.width = RenderText:sizeUtf8Text(0, Screen:getWidth(), face, w, true, bold).x
+                    table.insert(h_list, word_box)
+                end
+            end
+        end
+        if line:sub(-1) == "\n" then table.insert(h_list, { word = '\n', width = 0 }) end
+    end
+
+    -- @TODO check alg here 25.04 2012 (houqp)
+    -- @TODO replace greedy algorithm with K&P algorithm  25.04 2012 (houqp)
+    return self:_wrapGreedyAlg(h_list, width)
+end
+
+
+function StatusWidget:getStatisticsSettings()
+    local lastfile = G_reader_settings:readSetting("lastfile")
+    if lastfile then
+        local settings = DocSettings:open(lastfile)
+        if settings then
+            self.stats = settings:readSetting("stats")
+        end
+    end
+    if self.settings then
+        local currStats = self.settings:readSetting("stats")
```

Comment:
Another nitpicking, as a general coding style, we use `camelCase` for function names and `under_score` for variable names. It would nice if you can change `currStats` to curr_stats` :)